### DISCUSSION TITLE
Wireshark: negative timestamps

### DISCUSSION THREAD

**Discussion by NageshEranki** [bitcraze/discussions#1278] [Q&A] [answered]
Hi

My research requires me to monitor the radio channel for almost every flight. I currently run everything on an M3 MacBook Air, Sonoma 14.4.1. On some routine flights, some captured packets dissected using Wireshark have negative timestamps as in the figure below. The only way I am able to fix this is to repeatedly unplug and replug my Crazyradio PA and re-run the flights. Sometimes it takes 5 repetitions, sometimes 7, etc.

<img width="858" alt="Screenshot 2024-04-09 at 5 41 04 PM" src="https://github.com/bitcraze/discussions/assets/28665953/d839ce66-01b9-40af-afe1-982feae31d35">

If any other info is needed to help identify the bug, please let me know!

Regards,
Nagesh

**Comment by ToveRumar** [comment]
Hi!
Thanks for your question!

What setting do you have for your time column in Wireshark?
You could set it a couple of different ways, relative to start, delta between packets etc.

**Reply by NageshEranki** [reply]
Hi

i initially had it at the default setting of relative to first packet. I toyed around with the time format and tried delta between packets. In this Setting, exactly one packet has a negative delta. 

It’s like midway through my experiment, either the CF shifts the timestamps back in time or wireshark interprets the timestamps inaccurately.

One important clue is that I also regularly log CF variables and the timestamps from the CF of when I record them. I never had a problem plotting logged variables on the time axis: no funny time shifting business. 

I’m happy to share more details with you. 

Thanks
Nagesh

**Reply by ToveRumar** [reply]
Hi!
Ok, thanks!
This looks strange. 
Do you have some pcap file we could have a look at?

**Reply by NageshEranki** [reply]
Hi,

I'm not comfortable sharing PCAP files here. I sent it to your email for your reference.

Thanks,
Nagesh

**Reply by ToveRumar** [reply]
Hi! 
Great, thank you. We will look into it.

**Reply by ToveRumar** [reply, marked as answer]
Hi @NageshEranki!

Thank you for your patience on this issue. 

This issue is now fixes (see this [commit](https://github.com/bitcraze/crazyflie-lib-python/pull/454/commits/b875fca4a94cbd53367f7b24b2115343ad2c646e)) So please upgrade your client or introduce the same changes locally. 

If you have further problems with this please notify us by starting a new discussion on the topic or re-opening this one.
